app/src/main/java/com/frinder/frinder/model/User.java

```diff
@@ -135,20 +135,52 @@ public String toString() {
     }
 
     public static User fromJSON(JSONObject object) {
+        String id, name, email, gender, link, ageRange;
         User user = new User();
         try {
-            String id = object.getString(Constants.FACEBOOK_PROFILE_ID);
-            String name = object.getString(Constants.FACEBOOK_PROFILE_NAME);
-            String email = object.getString(Constants.FACEBOOK_PROFILE_EMAIL);
-            String gender = object.getString(Constants.FACEBOOK_PROFILE_GENDER);
-            String link = object.getString(Constants.FACEBOOK_PROFILE_LINK);
-            String ageRange = object.getString(Constants.FACEBOOK_PROFILE_AGE_RANGE);
+            id = object.getString(Constants.FACEBOOK_PROFILE_ID);
+
+            if (object.has(Constants.FACEBOOK_PROFILE_NAME)) {
```

Comment:
@mallikaviswas  Could we have a better way of doing this in id,name.....ageRange? 
1. either move the default values to the declaration?
2. even better would be no else and default value during declaration just have checking for !TextUtils.isEmpty(name) where ever used?